docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,277 @@
+* the named `slice` does not exist
+* if the `cpu` and `memory` controllers are not enabled on the specified slice
+
+### Kubernetes reserved
+
+The node may set aside a set of resources for Kubernetes components: 
+
+* `--kube-reserved=:` - A set of `ResourceName`=`ResourceQuantity` pairs that
+describe resources reserved for host daemons.
+
+The `kubelet` does not enforce `--kube-reserved` at this time, but the ability
+to distinguish the static reservation from observed usage is important for node accounting.
+
+This proposal asserts that all components that Kubernetes brings to the node are
+parented by a common `slice` in a separate part of the hierarchy than system-reserved
+resources.
+
+This proposal asserts that `kubernetes.slice` is the default slice associated with
+the `kubelet` and `kube-proxy` service units defined in the project.
+
+The `kubelet` will detect the parent `slice` from which it was launched to track
+kubernetes-reserved observed usage.
+
+If the `kubelet` is parented by a `slice` that is categorized as system-reserved,
+the `kubelet` will log a warning that kubernetes-reserved observed usage accounting is
+disabled, but the static reservation is still observed for purposes of reporting the
+allocable resources for the node.
+
+If the `kubelet` is launched directly from a terminal, it's most likely destination will
+be in a `scope` that is a child of `user.slice` as follows:
+
+`/sys/fs/cgroup/<controller>/user.slice/user-1000.slice/session-1.scope`
+
+In this context, the parent `scope` is what will be used to facilitate local developer
+debugging scenarios for tracking `kube-reserved` usage.
+
+### Kubernetes container runtime reserved
+
+This proposal asserts that the reservation of compute resources for any associated
+container runtime daemons is tracked by the operator under the system-reserved or
+kubernetes-reserved values at this time and any enforced limits are set by the
+operator specific to the container runtime.
+
+For future consideration, if the `kubelet` attempts to enforce `--kube-reserved`, this
+proposal asserts that the reservation associated with any particular container runtime
+should be split out at that time.  In a `systemd` environment, this proposal asserts that
+you cannot mandate the container runtime daemon is in the same `slice` as the Kubernetes
+components.
+
+**Docker**
+
+If the `kubelet` is configured with the `container-runtime` set to `docker`, the
+`kubelet` will detect the `service` associated with the `docker` daemon and use that
+to do local node accounting.  If an operator wants to impose runtime limits on the
+`docker` daemon to control resource usage, the operator should set those explicitly in
+the `service` unit that launches `docker`.  The `kubelet` will not set any limits itself
+at this time and will assume whatever budget was set aside for `docker` was included in
+either `--kube-reserved` or `--system-reserved` reservations.
```

Comment:
@jonboulle: On non-systemd distros where the kubelet is expected to setup cgroups for the daemons, it will continue to do so for the foreseeable future.

As for systemd distros, I'm suggesting letting the admins deal with node bootstrapping since there are multiple ways one would prefer to setup such nodes.